pkg/iam/iam.go

```diff
@@ -357,49 +350,60 @@ func (s *IAMService) applyAssumePolicyRole(roleName string, roleType string, par
 // attachInlinePolicy  will attach inline policy to the main IAM role
 func (s *IAMService) attachInlinePolicy(roleName string, roleType string, params interface{}) error {
 	l := s.log.WithValues("role_name", roleName)
-	i := &awsiam.ListRolePoliciesInput{
-		RoleName: aws.String(roleName),
-	}
+	tmpl := getInlinePolicyTemplate(roleType)
 
-	alreadyExists := false
+	policyDocument, err := generatePolicyDocument(tmpl, params)
+	if err != nil {
+		l.Error(err, "failed to generate inline policy document from template for IAM role")
+		return err
+	}
 
 	// check if the inline policy already exists
-	o, err := s.iamClient.ListRolePolicies(i)
-	if err == nil {
-		for _, p := range o.PolicyNames {
-			if *p == policyName(s.roleType, s.clusterName) {
-				alreadyExists = true
-				break
-			}
-		}
+	output, err := s.iamClient.GetRolePolicy(&awsiam.GetRolePolicyInput{
+		RoleName:   aws.String(roleName),
+		PolicyName: aws.String(policyName(s.roleType, s.clusterName)),
+	})
+	if err != nil && !IsNotFound(err) {
+		l.Error(err, "failed to fetch inline policy for IAM role")
+		return err
 	}
 
-	// add inline policy to the main IAM role if it do not exist yet
-	if !alreadyExists {
-		tmpl := getInlinePolicyTemplate(roleType)
-
-		policyDocument, err := generatePolicyDocument(tmpl, params)
+	if err == nil {
+		l.Info("output policy document", "policy_document", *output.PolicyDocument)
+		l.Info("new policy document", "policy_document", policyDocument)
```

Comment:
I think this might be too verbose for normal logs, should be hidden under Debug or something